.github/CODE_STANDARDS.md

```diff
@@ -0,0 +1,295 @@
+- [ ] Are there any performance concerns in the code?
+- [ ] Have new dependencies been appropriately vetted?
+- [ ] Does the code follow secure coding practices and avoid common vulnerabilities?
+- [ ] Is the code well(self)-documented, with clear variable, function naming?
+- [ ] Is the code efficient, readable, and free of code smells?
+- [ ] Have all review comments been addressed in a timely manner?
+- [ ] Do you understand the code and it's purpose?
+
+This checklist serves as a guide to both authors and reviewers to ensure a thorough and effective code review process.
+
+## 3. Contributing
+
+### 3.1 Pull Requests && Issues
+
+We have separate templates for Pull Requests and Issues. Please use them when creating a new PR or Issue.
+
+### 3.2 Conventional Commits & Pull Requests Naming
+
+#### 3.2.1 Overview
+
+In an effort to maintain clarity and coherence in our commit history, we are adopting the Conventional Commits style for all commit messages across our repositories. This uniform format not only enhances the readability of our commit history but also facilitates automated tools in generating changelogs and extracting valuable information effectively.
+
+#### 3.2.2 Structure
+
+Conventional Commits follow a structured format: `type(scope): description`, where:
+
+- `type`: Represents the nature of the commit (e.g., feat, fix, chore).
+- `scope`: Denotes the relevant module or issue.
+- `description`: Provides a brief explanation of the change.
+
+When introducing breaking changes, an `!` should be appended after the `type/scope`: `feat(#123)!: introduce a breaking change`.
+
+#### 3.2.3 Types
+
+- `feat`: Utilized when introducing a new feature to the codebase.
+- `fix`: Employed when resolving a bug or issue in the code.
+- `docs`: Designated for commits involving documentation changes, such as updating README files or adding comments.
+- `style`: Applied to commits focusing on code style and formatting, without altering the code's functionality.
+- `refactor`: Used for code changes that neither introduce new features nor fix bugs, but improve the code structure or design.
+- `test`: Assigned to commits pertaining to the addition, modification, or refactoring of tests.
+- `chore`: For changes related to build processes, local development, or other maintenance tasks.
+- `perf`: Employed when enhancing the performance of the codebase.
+- `revert`: Marked for commits that revert a previous change.
+- `ci`: Applied to changes concerning the Continuous Integration (CI) configuration or scripts.
+- `deps`: Used when updating or modifying dependencies.
+
+#### 3.2.4 Scope
+
+We have standardized the use of JIRA/GitHub issue numbers as the `scope` in commits within our team. This practice aids in easily tracing the origin of changes.
+
+In the absence of an existing issue for your changes, please create one in the client’s JIRA system. If the change is not client-related, establish a GitHub issue in the repository.
+
```

Comment:
What if someone doesn't have access to our JIRA? I think we can say here that in that case the Scope can also be a regular scope like `feat(User): create new addUser endpoint`, or `feat(api): create new addUser endpoint`.

And maybe, just maybe, we could say that if the scope is not easy to explain in one word, there can be no scope, but it's an exception and then it needs a good commit message. That is okay with the rules from https://www.conventionalcommits.org/en/v1.0.0/ .

WDYT?

Reply:
Any contributor doesn't have access to JIRA should use GitHub issue what is mentioned above